x-pack/plugins/ml/public/embeddables/map/layer/compatible_layer.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiAccordion,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { MapEmbeddable } from '@kbn/maps-plugin/public';
+import { JobDetails } from '../job_details';
+import type { LayerResult } from '../../../application/jobs/new_job/job_from_map';
+
+interface DropDownLabel {
+  label: string;
+  field: string;
+}
+
+interface Props {
+  embeddable: MapEmbeddable;
+  layer: LayerResult;
+  layerIndex: number;
+}
+
+export const CompatibleLayer: FC<Props> = ({ embeddable, layer, layerIndex }) => {
+  const [selectedSplitField, setSelectedSplitField] = useState<string | null>(null);
+
+  const splitFieldSelection: EuiComboBoxOptionOption[] = useMemo(() => {
+    const selectedOptions: EuiComboBoxOptionOption[] = [];
+    if (selectedSplitField) {
+      selectedOptions.push({
+        label: selectedSplitField,
+        field: selectedSplitField,
+      } as DropDownLabel);
+    }
+    return selectedOptions;
+  }, [selectedSplitField]);
+
+  const onSplitFieldChange = useCallback((selectedOptions: EuiComboBoxOptionOption[]) => {
+    const option = selectedOptions[0] as DropDownLabel;
+    if (typeof option !== 'undefined') {
+      setSelectedSplitField(option.field);
+    } else {
+      setSelectedSplitField(null);
+    }
+  }, []);
+  return (
+    <>
+      <EuiPanel paddingSize="m">
```

Comment:
Looks like there is an extra layer of nesting with the panels here, compared to the existing metric job one:

<img width="558" alt="image" src="https://user-images.githubusercontent.com/7405507/211578160-35520356-2db1-4810-8baa-9e54db2e4774.png">

<img width="509" alt="image" src="https://user-images.githubusercontent.com/7405507/211578333-5c528ea2-9893-4e5c-9651-d7364df25209.png">

Is it possible to keep the styling consistent?

Reply:
Ah - good catch - snuck in in the refactor. Removed so it's consistent with lens flyout in fff680a74dc070c51ba2b0ed0f5c9ec53e2b52f2